File: backend/app/domain/series.py
from pydantic import BaseModel, Field


class SeriesResponse(BaseModel):
    id: str
    title: str
    slug: str


class EpisodeResponse(BaseModel):
    id: str
    series_id: str
    season_number: int = Field(ge=1)
    episode_number: int = Field(ge=1)
    episode_order: int = Field(ge=1)
    code: str
    title: str
    visible_from_order: int = Field(ge=1)
    # D-21 additive display shape (07-03): the API always returns the
    # already-masked value in `title` when a boundary is applied (backward
    # compatible), and additionally exposes the masked display_title plus the
    # unlock/view flags so the frontend can render watched/current/locked
    # states without any client-side masking logic (D-08).
    display_title: str | None = None
    is_unlocked: bool | None = None
    is_current_view: bool | None = None

File: frontend/src/types/series.ts
// Mirrors backend/app/domain/series.py field-for-field.

export type SeriesResponse = {
  id: string
  title: string
  slug: string
}

export type EpisodeResponse = {
  id: string
  series_id: string
  season_number: number
  episode_number: number
  episode_order: number
  code: string
  title: string
  visible_from_order: number
  // D-21 additive display fields (07-03): the backend always returns the
  // already-masked value in `display_title` when a boundary is applied; the
  // legacy `title` field keeps the raw title for backward compatibility.
  display_title?: string | null
  is_unlocked?: boolean | null
  is_current_view?: boolean | null
}
